### PR TITLE
Add WithItems() to TransactWriteItems

### PR DESCRIPTION
**Pull request by OmarElgabry**
It is very useful to test what actually really goes into `TransactWriteItems()`

### CODE REVIEW

**Review by asad-awadia — APPROVED** [review]

**Comment by gusaul** [comment]
please resolve the conflict

File: transact_write_items.go [hunk 1]
```diff
@@ -52,32 +64,5 @@ func (e *MockDynamoDB) TransactWriteItems(input *dynamodb.TransactWriteItemsInpu
...
 
 func (e *MockDynamoDB) TransactWriteItemsWithContext(ctx aws.Context, input *dynamodb.TransactWriteItemsInput, opts ...request.Option) (*dynamodb.TransactWriteItemsOutput, error) {
-	if len(e.dynaMock.TransactWriteItemsExpect) > 0 {
-		x := e.dynaMock.TransactWriteItemsExpect[0] //get first element of expectation
-
-		foundTable := false
-
-		if x.table != nil {
-			for _, item := range input.TransactItems {
-				if (item.Update != nil && x.table == item.Update.TableName) ||
```
Comment (OmarElgabry) [inline transact_write_items.go:62]: Not sure if you guys reviewed this or not, but it seems like if one item has the correct table name, but the rest don't, the `foundTable` will still be set to true! And it is not one table, TransactWriteItems can have items that target different tables.

File: transact_write_items.go [hunk 2]
```diff
@@ -13,35 +15,45 @@ func (e *TransactWriteItemsExpectation) Table(table string) *TransactWriteItemsE
...
+					(item.Put != nil && x.table != item.Put.TableName) ||
+					(item.Delete != nil && x.table != item.Delete.TableName) {
+					return nil, fmt.Errorf("Expect table %s not found", *x.table)
 				}
 			}
 
-			if foundTable == false {
-				return nil, fmt.Errorf("Expect table %s not found", *x.table)
+			// compare transact write item - each item also contains the table name
+			if !reflect.DeepEqual(x.items[i], item) {
```
Comment (OmarElgabry) [inline transact_write_items.go]:
```suggestion
			if x.items[i] != nill && !reflect.DeepEqual(x.items[i], item) {
```

File: transact_write_items.go [hunk 3]
```diff
@@ -13,35 +15,45 @@ func (e *TransactWriteItemsExpectation) Table(table string) *TransactWriteItemsE
...
+					(item.Put != nil && x.table != item.Put.TableName) ||
+					(item.Delete != nil && x.table != item.Delete.TableName) {
+					return nil, fmt.Errorf("Expect table %s not found", *x.table)
 				}
 			}
 
-			if foundTable == false {
-				return nil, fmt.Errorf("Expect table %s not found", *x.table)
+			// compare transact write item - each item also contains the table name
+			if x.items[i] != nill && !reflect.DeepEqual(x.items[i], item) {
```
Comment (OmarElgabry) [inline transact_write_items.go]:
```suggestion
			if x.items[i] != nil && !reflect.DeepEqual(x.items[i], item) {
```

**Review by gusaul — APPROVED** [review]

**Pull request merged**